java/algorithm/src/main/java/org/apache/arrow/algorithm/move/VectorDataCopier.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.move;
+
+import org.apache.arrow.vector.BaseFixedWidthVector;
+import org.apache.arrow.vector.BaseVariableWidthVector;
+import org.apache.arrow.vector.BitVectorHelper;
+
+import io.netty.buffer.ArrowBuf;
+import io.netty.util.internal.PlatformDependent;
+
+/**
+ * Utilities for moving data between vectors.
+ */
+public class VectorDataCopier {
+
+  /**
+   * Copy data for fixed-width vectors.
+   * @param srcVector the source vector.
+   * @param srcIndex the data index in the source vector.
+   * @param dstVector the destination vector.
+   * @param dstIndex the data index in the destination vector.
+   * @param typeWidth the width of the data type.
+   */
+  public static void fixedWidthDataCopy(
```

Comment:
would it make sense to add a swap method to FixedWidthVector classes instead of doing this externally?  For in-place sorting I think you probably want a swap operation instead anyways?

Reply:
Good suggestion. Swap is definitively a useful feature. 
However, it is more difficult to implement it, because it requires extra memory space. In addition, it is not used in the code so far. So can we do it in the future by opening a separate JIRA?

Reply:
If it is needed we can add it.  I also meant to ask is this more appropriate as a method on BaseFixedWidthVector?

Reply:
As you suggested, there are two ways to implement the feature:
1. As a method on BaseFixedWidthVector
2. In a separate class in the algorithm module.

Generally, I prefer solution 2, because
1. Sometimes, the swap may involve two vectors, and 
2. Swap requires extra memory, and a separate class makes it easier to manage the extra memory.

Reply:
@emkornfield thanks for the good point. I agree that implementing the copyTo as a member function is a better choice.

However, currently each vector has its own implementation of the copyFrom method, which performs similar functionality. So the current problem is to extract these copyFrom methods to the super class, so we can use it in the sorter.

This involves substantial changes, so I think we need a separate JIRA for it. I have opened one (ARROW-5778). Let's hold this PR until that one is resolved.

Reply:
SGTM